### PR TITLE
[SDK-246] Add AML check

### PR DESCRIPTION
**Pull request by vassyz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -208,6 +181,51 @@ end
...
+
+```ruby
+require 'yoti'
+
+Yoti.configure do |config|
+  config.client_sdk_id = ENV['YOTI_CLIENT_SDK_ID']
+  config.key_file_path = ENV['YOTI_KEY_FILE_PATH']
+end
+
+aml_address = Yoti::AmlAddress.new('GBR', 'E1 6DB')
```
Comment (qzagarese) [inline README.md]: I think we should remove the postcode from this example as it is not needed for UK.
It is misleading. We should then propagate the change to other readmes

File: lib/yoti/http/payloads/aml_profile.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    def profile_valid
+      !@given_names.to_s.empty? && !@family_name.to_s.empty? && address_valid
+    end
+
+    def address_valid
+      @address.is_a?(AmlAddress) && @address.country.to_s.length == 3
+    end
+
+    def usa_valid
+      @address.country != 'USA' || (!@ssn.nil? && !@address.post_code.to_s.empty?)
```
Comment (qzagarese) [inline lib/yoti/http/payloads/aml_profile.rb]: should this only contain the USA specific validation logic and then be used as `unless profile_valid && usa_valid`
(not sure whether syntax supports it)

File: lib/yoti/http/payloads/aml_profile.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  # Manages the AML check Profile object
+  class AmlProfile
+    def initialize(given_names, family_name, aml_address, ssn = nil)
+      @given_names = given_names
+      @family_name = family_name
+      @ssn = ssn
+      @address = aml_address
+
+      raise AmlError, 'The AML request requires given names, family name and an ISO 3166 3-letter code.' unless profile_valid
+      raise AmlError, 'Request for USA require a valid SSN and post code.' unless usa_valid
```
Comment (echarrod) [inline lib/yoti/http/payloads/aml_profile.rb]: Isn't it called Zip code in the US? Or are we just going to say postcode everywhere for consistency? 

(If we do stick with postcode, it is one word I believe)

File: spec/yoti/http/payloads/aml_profile_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    context 'with missing values' do
+      it 'raises Yoti::AmlError' do
+        error = 'The AML request requires given names, family name and an ISO 3166 3-letter code.'
+        expect { Yoti::AmlProfile.new(nil, '', []) }.to raise_error(Yoti::AmlError, error)
+      end
+    end
+
+    context 'with an invalid USA address values' do
+      it 'raises Yoti::AmlError' do
+        error = 'Request for USA require a valid SSN and post code.'
```
Comment (echarrod) [inline spec/yoti/http/payloads/aml_profile_spec.rb]: ditto

File: lib/yoti/http/payloads/aml_profile.rb (hunk 2 above)
Comment (vassyz) [reply lib/yoti/http/payloads/aml_profile.rb]: I'm not sure I understand. Does it not contain only USA specific validation?

File: lib/yoti/http/payloads/aml_profile.rb (hunk 3 above)
Comment (vassyz) [reply lib/yoti/http/payloads/aml_profile.rb]: We'll use postcode as it is the more generic.

File: lib/yoti/http/payloads/aml_profile.rb (hunk 2 above)
Comment (qzagarese) [reply lib/yoti/http/payloads/aml_profile.rb]: Forget it, I misread ;-;

File: lib/yoti/http/payloads/aml_profile.rb (hunk 3 above)
Comment (echarrod) [reply lib/yoti/http/payloads/aml_profile.rb]: I was just thinking that the only time we'll be showing this error will be for users in the US - so zip code might be better. But I guess if we call it zip code here and postcode elsewhere it'll be inconsistent

File: lib/yoti/http/payloads/aml_profile.rb (hunk 2 above)
Comment (vassyz) [reply lib/yoti/http/payloads/aml_profile.rb]: I think you're right though, it looks a bit confusing. I'll rewrite it.

**Comment by echarrod** [comment]
`examples/aml_ckeck/app.rb` -> typo in `aml_ckeck` folder

**Comment by sidibos** [comment]
LGTM

**Pull request merged**